clusterctl/cmd/register.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+type RegisterOptions struct {
+	RegistryEndpoint string
+}
+
+var ro = &RegisterOptions{}
+
+var registerCmd = &cobra.Command{
+	Use:   "register",
```

Comment:
Should register also follow VERB NOUN? ie. "register cluster" instead of just "register"?

Reply:
Whenever register is implemented, it can be branched out. For now, I'm leaving it as is.

Reply:
I do not think this is the right approach. 

Either we structure it as it is supposed to be structured ie. VERB NOUN or we leave it out entirely and when it is implemented, it is added with the right structure.

Reply:
I have decided to remove it and leave it up to implementer to add this cmd as well.